pkg/validation/spec/swagger.go

```diff
@@ -46,6 +46,21 @@ func (s Swagger) MarshalJSON() ([]byte, error) {
 	return swag.ConcatJSON(b1, b2), nil
 }
 
+func (s Swagger) MarshalNext() ([]byte, error) {
+	return jsonv2.Marshal(s)
+}
+
+// MarshalJSON marshals this swagger structure to json
+func (s Swagger) MarshalNextJSON(opts jsonv2.MarshalOptions, enc *jsonv2.Encoder) error {
+	var x struct {
+		Extensions
+		SwaggerProps
+	}
+	x.Extensions = s.Extensions
+	x.SwaggerProps = s.SwaggerProps
```

Comment:
curious if there's any alloc difference between making the fields of these unified types pointers so they only need to take the address of the subfield we're marshaling, e.g.

```go
var x struct {
  *Extensions
  *SwaggerProps
}
x.Extensions = &s.Extensions
x.SwaggerProps = &s.SwaggerProps
```

(if that improves allocs, the same comment might apply other places we're doing this)

Reply:
I don't know what to think:
```
name                                      old time/op    new time/op    delta
SwaggerSpec_ExperimentalMarshal/jsonv2-8    31.5ms ± 1%    31.6ms ± 1%     ~     (p=0.743 n=8+9)

name                                      old alloc/op   new alloc/op   delta
SwaggerSpec_ExperimentalMarshal/jsonv2-8    22.4MB ± 0%    19.1MB ± 0%  -14.70%  (p=0.000 n=8+9)

name                                      old allocs/op  new allocs/op  delta
SwaggerSpec_ExperimentalMarshal/jsonv2-8     91.3k ± 0%    105.7k ± 0%  +15.73%  (p=0.000 n=7+10)
```

Reply:
eh, probably not worth it then

Reply:
🤷